extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/IntervalUtils.java

```diff
@@ -61,4 +64,23 @@ public static List<Interval> difference(final List<Interval> list1, final List<I
 
     return retVal;
   }
+
+  /**
+   * This method checks if the provided interval is aligned by the granularity or is an instance of {@link Intervals#ETERNITY}
+   * ALL granularity isn't aligned to any interval, however this method is defines that ALL granularity matches
+   * an interval with boundary ({@code DateTimes.MIN}, {@code DateTimes.MAX})
+   * This is used to check if the granularity allocation made by the overlord is the same as the one requested in the
+   * SQL query
+   */
+  public static boolean isEternityOrDoesIntervalAlignWithGranularity(
```

Comment:
Nit: you need not mention the part about eternity as an eternity interval does satisfy the definition of alignment with `AllGranularity`.

So this method can just be called `isAligned` or something to reflect the underlying usage of `granularity.isAligned`.